db/migrate/20200607025146_add_ancestry_to_vm.rb

```diff
@@ -0,0 +1,112 @@
+class AddAncestryToVm < ActiveRecord::Migration[5.2]
+  class VmOrTemplate < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    self.table_name = 'vms'
+    has_many :all_relationships, :class_name => "AddAncestryToVm::Relationship", :dependent => :destroy, :as => :resource
+    include ActiveRecord::IdRegions
+  end
+
+  class Relationship < ActiveRecord::Base
+  end
+
+  #
+  # build sql that converts relationship ancestry with the id of the relationship to model-side ancestry with the id of the model
+  #
+
+  # def ancestry_resource_ids(relationship, model_class_name, id_range)
+  #   Relationship.where(
+  #     :relationship => relationship,
+  #     :resource_type => model_class_name,
+  #     :resource_id => id_range
+  #   ).where.not(:ancestry => nil).flat_map do |a_rels|
+  #     a_rels.ancestry.split('/').each_with_index.map { |rel_id, indx| [a_rels.resource_id, rel_id, indx] }
+  #   end
+  # end
+
+  def ancestry_resource_ids(relationship, model_class_name, id_range)
```

Comment:
I know we had to put in ruby code that mimics the sql because it is just too much.

but (at least now when the code is fresh in my mind) like the end result.

getting this into a migration routine may be nice - not sure if we want something like this in ancestry.
Definitely want a better way to populate ancestry based upon sql. (sorry tangent)
^ ignore the ancestry comment. want to keep the thought out there but  don't want us to move towards that right now

Reply:
`but (at least now when the code is fresh in my mind) like the end result.`
Are you missing a verb? Is this saying that you do like the end result here now, or that you do not?

Reply:
sorry, missed a noun. I **do** like the end result

Reply:
um, I missed the "I" which is a pro noun.

I guess I then added the do to answer your question of do or do not (that is the question).

So I stand by my noun statement.